org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/RuntimeEnvironment.java

```diff
@@ -137,11 +139,14 @@ public URL getJavadocURL() {
 			url = new URL(javadoc);
 		} catch (MalformedURLException e) {
 			File file = new File(javadoc);
-			if (file.exists()) {
+			if (file.exists() && file.isAbsolute()) {
 				try {
-					return JDTUtils.toURI(javadoc).toURL();
-				} catch (MalformedURLException e1) {
-					JavaLanguageServerPlugin.logException(e.getMessage(), e);
+					URI uri = new URI("file", javadoc, null);
+					uri = new URI(ResourceUtils.fixURI(uri));
+					return uri.toURL();
+				} catch (MalformedURLException | IllegalArgumentException | URISyntaxException e1) {
+					JavaLanguageServerPlugin.logInfo("Invalid javadoc: " + javadoc);
```

Comment:
I think with this logInfo, it would get logged twice (there's one outside the catch as well on 149). I would just remove this one and let the one below get used.

Reply:
Fixed.

Reply:
Merged.